CMakeLists.txt

```diff
@@ -376,3 +377,32 @@ add_custom_target(
   COMMAND ${CMAKE_COMMAND} -DCOMPONENT=viam-cpp-sdk_examples -P ${CMAKE_BINARY_DIR}/cmake_install.cmake
   DEPENDS all
 )
+
+add_custom_target(
+  install-tests
+  COMMAND ${CMAKE_COMMAND} -DCOMPONENT=viam-cpp-sdk_tests -P ${CMAKE_BINARY_DIR}/cmake_install.cmake
+  DEPENDS all
+)
+
+enable_testing()
+function(add_boost_test SOURCE_FILE_NAME)
```

Comment:
I recommend to create a home for the inevitable accumulation of helper functions that this project will have over time rather than placing them in the top level. Make a new top-level directory called `cmake`, and add a file called `viamcpp_utils.cmake`. Then put this function in it. Then up in the top-level `CMakeLists.txt` you can add the `cmake` directory to the module search path:
```
set(CMAKE_MODULE_PATH
   ${CMAKE_MODULE_PATH}
   ${PROJECT_SOURCE_DIR}/cmake
)
```

Then you can `import(viamcppsdk_utils)` to pick up the function.

Also, rename the function to something like `viamcppsdk_add_boost_test` to namespace it.